core/src/consensus.rs

```diff
@@ -569,13 +569,36 @@ impl Tower {
                     return SwitchForkDecision::SameFork;
                 }
 
-                assert!(
-                    !last_vote_ancestors.contains(&switch_slot),
-                    "Should never consider switching to slot ({}), which is ancestors({:?}) of last vote: {}",
-                    switch_slot,
-                    last_vote_ancestors,
-                    last_voted_slot
-                );
+                if last_vote_ancestors.contains(&switch_slot) {
+                    if !self.is_stray_last_vote() {
+                        panic!(
+                            "Should never consider switching to slot ({}), which is ancestors({:?}) of last vote: {}",
+                            switch_slot,
+                            last_vote_ancestors,
+                            last_voted_slot
+                        );
+                    } else {
+                        // This peculiar corner handling is needed mainly for a tower which is newer than
```

Comment:
@carllin hehe, this was quite hard... ;)

Reply:
thanks for the detective work! Was this caused after a deleted/corrupted ledger?

Reply:
basically yes. :)